src/Features/LanguageServer/Microsoft.CommonLanguageServerProtocol.Framework.Example/Microsoft.CommonLanguageServerProtocol.Framework.Example.csproj

```diff
@@ -11,6 +11,7 @@
 
   <ItemGroup Label="Package References">
     <PackageReference Include="Microsoft.Extensions.DependencyInjection" Version="$(MicrosoftExtensionsDependencyInjectionVersion)" />
+    <PackageReference Include="Microsoft.Bcl.AsyncInterfaces" Version="$(MicrosoftBclAsyncInterfacesVersion)" />
```

Comment:
You shouldn't reference this package on .NETCoreApp.

Reply:
I don't want to add this but I get [this](https://github.com/dotnet/roslyn/runs/20103346733)
```
.dotnet/sdk/8.0.100/Microsoft.Common.CurrentVersion.targets(2382,5): error MSB3277: (NETCORE_ENGINEERING_TELEMETRY=Build) Found conflicts between different versions of "Microsoft.Bcl.AsyncInterfaces" that could not be resolved.
There was a conflict between "Microsoft.Bcl.AsyncInterfaces, Version=7.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51" and "Microsoft.Bcl.AsyncInterfaces, Version=8.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51".
    "Microsoft.Bcl.AsyncInterfaces, Version=7.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51" was chosen because it was primary and "Microsoft.Bcl.AsyncInterfaces, Version=8.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51" was not.
    References which depend on "Microsoft.Bcl.AsyncInterfaces, Version=7.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51" [/mnt/vss/_work/1/s/.packages/microsoft.bcl.asyncinterfaces/7.0.0/lib/netstandard2.1/Microsoft.Bcl.AsyncInterfaces.dll].
        /mnt/vss/_work/1/s/.packages/microsoft.bcl.asyncinterfaces/7.0.0/lib/netstandard2.1/Microsoft.Bcl.AsyncInterfaces.dll
          Project file item includes which caused reference "/mnt/vss/_work/1/s/.packages/microsoft.bcl.asyncinterfaces/7.0.0/lib/netstandard2.1/Microsoft.Bcl.AsyncInterfaces.dll".
            /mnt/vss/_work/1/s/.packages/microsoft.bcl.asyncinterfaces/7.0.0/lib/netstandard2.1/Microsoft.Bcl.AsyncInterfaces.dll
    References which depend on "Microsoft.Bcl.AsyncInterfaces, Version=8.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51" [].
        /mnt/vss/_work/1/s/artifacts/bin/Microsoft.CodeAnalysis.LanguageServer.Protocol/Debug/netstandard2.0/Microsoft.CodeAnalysis.LanguageServer.Protocol.dll
          Project file item includes which caused reference "/mnt/vss/_work/1/s/artifacts/bin/Microsoft.CodeAnalysis.LanguageServer.Protocol/Debug/netstandard2.0/Microsoft.CodeAnalysis.LanguageServer.Protocol.dll".
            /mnt/vss/_work/1/s/artifacts/bin/Microsoft.CodeAnalysis.LanguageServer.Protocol/Debug/netstandard2.0/Microsoft.CodeAnalysis.LanguageServer.Protocol.dll
        /mnt/vss/_work/1/s/artifacts/bin/Microsoft.CommonLanguageServerProtocol.Framework/Debug/netstandard2.0/Microsoft.CommonLanguageServerProtocol.Framework.dll
          Project file item includes which caused reference "/mnt/vss/_work/1/s/artifacts/bin/Microsoft.CommonLanguageServerProtocol.Framework/Debug/netstandard2.0/Microsoft.CommonLanguageServerProtocol.Framework.dll".
            /mnt/vss/_work/1/s/artifacts/bin/Microsoft.CommonLanguageServerProtocol.Framework/Debug/netstandard2.0/Microsoft.CommonLanguageServerProtocol.Framework.dll
            /mnt/vss/_work/1/s/artifacts/bin/Microsoft.CodeAnalysis.LanguageServer.Protocol/Debug/netstandard2.0/Microsoft.CodeAnalysis.LanguageServer.Protocol.dll
```

From `project.assets.json` file I see it the project is referencing the `7.0.0`  BCL in net6.0 TFM. My guess is it might be introduced by other packages used in project. And Roslyn is referencing `8.0.0` BCL in other referenced projects so it causes the conflict.
This is by far the easiest way I found to solve the issue.

Reply:
I believe the request here is to add it to a conditional item group:

```xml
<ItemGroup Condition="'$(TargetFramework)' == 'net472'">
  <PackageReference ... />
</ItemGroup>
```

Reply:
Looks like the following packages bring M.Bcl.AsyncInterfaces in on .NETCoreApp:
- Microsoft.VisualStudio.Threading/17.7.35
- Nerdbank.Streams/2.10.69
- StreamJsonRpc/2.17.8

Those packages should be updated to not reference M.Bcl.AsyncInterfaces. I submitted https://github.com/microsoft/vs-threading/pull/1269 for the M.VS.Threading packages.

Reply:
I am thinking about this. The main purpose of this PR is Roslyn also ships a few dlls, which needs to match the dlls version in msbuild. (e.g SystemCollectionsImmutableVersion and SystemReflectionMetadataVersion)
Even other packages in VS move to the new version (like `BCL`) it might be fine since we don't ship that dlls. We can wait for the upstream packages to update and later update all other stuff here. 
So here we just want to find a sweet point without too many other tricks. Testing now..